### PR TITLE
Replace hackathon Python stack with Go v1alpha1 operator

### PR DESCRIPTION
**Pull request by MFS-code**
## Summary

- Replace the Phase 1 Python/kopf hackathon stack with a Go + kubebuilder v1alpha1 operator (`Agent` + `AgentRun`).
- Add reconcilers, pod builder, provider credential wiring, kind install/e2e scripts, and echo + python-anthropic runtime images.
- Retire hackathon artifacts from main; preserve the full historical tree (including `CONCEPT.md`, TUI, v1 CRD) on `deprecated/hackathon-python`.

## Test plan

- [x] `make test` (unit + envtest reconciler tests)
- [ ] `./scripts/install-go-kind.sh && ./scripts/e2e-kind.sh` on a fresh kind cluster
- [ ] Standalone `AgentRun` completes with `.status.result`
- [ ] `Service` `Agent` re-casts after Pod deletion

### CODE REVIEW

File: internal/controller/agentrun_controller.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	return r.syncPodObservation(ctx, &run, pod, podName)
+}
+
+func (r *AgentRunReconciler) reconcileMissingPod(ctx context.Context, run *kontextv1alpha1.AgentRun, podName string) (ctrl.Result, error) {
+	if status.IsTerminalPhase(run.Status.Phase) {
+		return ctrl.Result{}, nil
+	}
+
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply internal/controller/agentrun_controller.go:73]: Fixed in f5036f0. Active runs now sync pod observation before returning the wallclock requeue, with regression coverage for the Running transition.

File: internal/runtimepolicy/provider.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+			SecretKey:         "AWS_ACCESS_KEY_ID",
+			DefaultSecretName: "kontext-bedrock",
+		},
+	},
+	"fake":          {keyless: true},
+	"replay":        {keyless: true},
+	"echo":          {keyless: true},
+	"deterministic": {keyless: true},
+}
+
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply internal/runtimepolicy/provider.go]: Fixed in f5036f0. Provider credential wiring now supports multiple secret-backed env vars, and Bedrock injects both required static credential fields.

File: internal/podbuilder/podbuilder.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+func PodNameForRun(runName string) string {
+	safe := invalidNameChars.ReplaceAllString(strings.ToLower(runName), "-")
+	safe = strings.Trim(safe, "-")
+	if safe == "" {
+		safe = "run"
+	}
+	if len(safe) > 56 {
+		safe = safe[:56]
+	}
+	return fmt.Sprintf("run-%s", safe)
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply internal/podbuilder/podbuilder.go:188]: Fixed in f5036f0. Pod names are trimmed again after truncation, with a boundary-case regression test.

File: internal/controller/agent_controller.go [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	}
+	return delay
+}
+
+func timeSinceCompletion(run *kontextv1alpha1.AgentRun) time.Duration {
+	if run.Status.CompletionTime != nil {
+		return time.Since(run.Status.CompletionTime.Time)
+	}
+	return time.Duration(defaultBackoffInitial) * time.Second
+}
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply internal/controller/agent_controller.go:236]: Fixed in f5036f0. A missing completion timestamp now means zero elapsed time, preserving the configured initial backoff; the reconciler behavior is covered by a regression test.

File: api/v1alpha1/agent_types.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+type AgentSpec struct {
+	Mode AgentMode `json:"mode"`
+
+	Runtime RuntimeSpec `json:"runtime"`
+
+	Goal         string `json:"goal,omitempty"`
+	GoalTemplate string `json:"goalTemplate,omitempty"`
+
+	Provider string   `json:"provider,omitempty"`
+	Model    string   `json:"model"`
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply api/v1alpha1/agent_types.go]: Fixed in 8d83382. `Agent.Spec.Model` now carries `MinLength=1`, matching the constraint on `AgentRunSpec.Model`, so an empty model is rejected at admission instead of looping in the reconciler.

File: runtimes/python-anthropic/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
+FROM python:3.13-slim
+
+WORKDIR /app
+
+COPY pyproject.toml README.md ./
+COPY src ./src
+
+RUN pip install --no-cache-dir .
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply runtimes/python-anthropic/Dockerfile:8]: Fixed in 0b62b02. The runtime is now a self-contained build context: the Anthropic runner from the retired stack was restored under `runtimes/python-anthropic/src/kontext_runtime/` with its own `pyproject.toml` exposing only the `kontext-runner` entrypoint (no operator or TUI packages). Verified `docker build` succeeds and the entrypoint resolves inside the image.

**Comment by MFS-code** [comment]
Addressed the two remaining defects from the review summary in 404d5e7: the Anthropic runner now sends the configured token budget to the API instead of silently capping it at 1024 (the constant remains only as the default when no budget is set), and pod observation reports `Waiting.Reason` — with `Waiting.Message` appended when present — so states like `ContainerCreating` no longer produce an empty waiting message. Both are covered by tests.

File: internal/controller/agent_controller.go [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+		run = &existing
+	}
+
+	nextStatus := kontextv1alpha1.AgentStatus{
+		CurrentRunName:     run.Name,
+		LastRunName:        run.Name,
+		RunsCreated:        agent.Status.RunsCreated,
+		Restarts:           agent.Status.Restarts,
+		ObservedGeneration: agent.Generation,
+	}
```
[the remark this answers is not in the document]
Comment (MFS-code) [reply internal/controller/agent_controller.go:132]: Fixed in c0ee45a. Both status-write paths now preserve history: the recast path records the replaced run's name in `LastRunName`, and the active-run path carries the existing `LastRunName` forward instead of overwriting it with the current run. Covered by an extended recast regression test that checks the field across two reconciles. The same commit also addresses the two table-level notes: Azure/azure-openai now wire `AZURE_OPENAI_ENDPOINT` alongside the API key, and `parse_duration_seconds` parses compound Go durations like `1h30m`.
